pixeltable/catalog/table_version.py

```diff
@@ -142,22 +157,23 @@ class IndexInfo:
 
     def __init__(
         self,
-        id: UUID,
+        key: TableVersionKey,
         tbl_md: schema.TableMd,
         version_md: schema.TableVersionMd,
-        effective_version: int | None,
         schema_version_md: schema.TableSchemaVersionMd,
         mutable_views: list[TableVersionHandle],
         base_path: 'TableVersionPath' | None = None,
         base: TableVersionHandle | None = None,
     ):
+        assert key.effective_version is None or key.anchor_tbl_id is None
```

Comment:
Can we put these in the key c'tor?

Reply:
`NamedTuple` doesn't give user-level control over the constructor; it's literally just a tuple with types.